### PR TITLE
Add singular form for verbose time description

### PR DESCRIPTION
**Pull request by mkchoi212**
When using `VLCTime.verboseDescription`,

1 **minutes** 20 seconds is now 1 **minute** 20 seconds

### CODE REVIEW

**Review by fkuehne — APPROVED** [review]
Patch itself looks good to me!

What I’m wondering about - didn’t newer versions of macOS / iOS introduce convenience code to print such user visible strings in all locales? Can we use that?

**Comment by mkchoi212** [comment]
@fkuehne Are you referring to [this](http://macoscope.com/blog/effective-localization-when-working-with-language-plural-rules/)?

It's cool but I don't see any `Localizable.strings` files in the project. Am I missing something?

**Comment by carolanitz** [comment]
awesome!!! This is exactly the reason why we need the Testsuite. To find all those tiny bugs and fix them along the way and make sure that this doesn't break in the future 🙌

**Review by carolanitz — APPROVED** [review]

**Comment by carolanitz** [comment]
@fkuehne I think you might be talking about using the Date components Api. We should definitely switch to that here and not try to reimplement the logic. 
I'm not sure if there are differences in time formats by country but if, we don't handle that. 
Also we would be wrong for 1 hour and 1 second.
The code should get more consize by using date components but make sure to only instantiate the object once for the entire class since it's very expensive to create.

https://developer.apple.com/documentation/foundation/datecomponents

**Comment by fkuehne** [comment]
Yes, that’s exactly the API I had in mind. Since we will require macOS 10.10 for VLCKit 4, it will be a good idea to switch to it.

**Comment by mkchoi212** [comment]
One question! When you have 1:00:30, do we want this to be "1 hour 30 seconds" or just "1 hour"?

**Comment by fkuehne** [comment]
I would prefer “1 hour 30 seconds” to be as precise as possible. Custom strings can be created easily by the client app if desired.

**Comment by mkchoi212** [comment]
I also noticed "1 hour 20 minutes 30 seconds" is represented as "1 hour 20 minutes". Is this desired behavior?

**Comment by carolanitz** [comment]
Why don't you take a look at what the nsdateformatter will give you by default? The rest just comes down to personal preference  ¯\_(ツ)_/¯  and like felix says if someone wants something else they can easily adapt it

**Comment by mkchoi212** [comment]
Ok cool. I'll go with the default format. I was just wondering cause I thought there was some particular reason behind things :D

File: Sources/VLCTime.m [hunk 1]
```diff
@@ -24,7 +25,18 @@
...
+- (void)initDateComponents;
+@end
+
 @implementation VLCTime
+static NSDateComponents *components = nil;
+
+- (void)initDateComponents
+{
+    if (!components)
+        components = [[NSDateComponents alloc] init];
```
Comment (carolanitz) [inline Sources/VLCTime.m]: turns out I've been mixing up the expensiveness of NSDateFormatter with NSDateComponents.
You can use this one actually directly.

**Review by carolanitz — APPROVED** [review]
the rest looks good. I would add a couple of examples in the description and update the News to note that we adjusted this method . As reference for the description I would point to - (NSString *)stringFromDate:(NSDate *)fromDate toDate:(NSDate *)toDate; and I think this can be merged then

File: Sources/VLCTime.m (hunk 1 above)
Comment (mkchoi212) [reply Sources/VLCTime.m]: 👍

**Comment by carolanitz** [comment]
I just tested it and it works great but the localization for remaining doesn't work. I will merge it anyway because it's improvement.
For german you get 1 Stunde 33 Minuten und 41 Sekunden remaining that should be fixed at some point

**Comment by carolanitz** [comment]
this was merged

**Pull request closed without merging**